docs/writing_a_plugin.md

```diff
@@ -76,40 +97,68 @@ Remember that Qlik only takes the first column of data sent back from the plugin
+
+### Metadata sent from the plugin to Qlik
+The SSE plugin can also send metadata back to Qlik containing information about the result.
+
+#### Cache control
 The default behavior of the Qlik engine is to cache results from computations in order to reduce the workload and speed up response time. This is valid as well for computations performed by SSE plugins. For some cases caching might be undesirable and it is possible to toggle it off per request. When `EvaluateScript` or `ExecuteFunction` are called the http header key:value pair `qlik-cache:no-store` can be set in either the initial or trailing header data of the response. This functionality is demonstrated in the [Hello world example](../examples/python/HelloWorld/README.md#nocache-function).
 
 Caching is automatically turned off in Qlik if a request fails due to the communication failure or an exception is raised during a call.
 
-## Using SSE plugin expressions in a Qlik app/document
+#### `TableDescription`
+This header must be sent as initial metadata, that is _before_ the plugin starts sending back the response data to Qlik. The information is only handled by Qlik within a data load, more precisely, in a `LOAD ... EXTENSION ...` statement. By sending the header, you can set metadata for the returned table such as __names__, __data types__, __tags__ etc.
+
+There are a few things to keep in mind when sending a `TableDescription`:
+* The number of `FieldDescriptions` in the message must match the number of returned fields to Qlik. Otherwise an error is raised.
+* If a message field is not set, the default value according to protobuf will be used.
+  * For example, if the datatype (int) is not set for a field (FieldDescription), the default value is 0, and the values of the returned field will be parsed in Qlik as strings.
+  * An exception is the _field name_, which will be set by default to _Field{i}_ for the _i:th_ field returned to Qlik if no other name was set.
+
+See more details in the [TableDescription](SSE_Protocol.md#qlik.see.TableDescription) section in the protocol documentation and in the example [FullScriptSupport using Pandas](../examples/python/FullScriptSupport_Pandas/README.md) written in Python where the header can be modified directly from the script written in Qlik.
+
+## Using SSE expressions in a Qlik app/document
+
+It is possible to make an SSE call from both the load script and from a chart expression.  
 
-It is possible to make an SSE call from the load script or from a chart expression. The syntax of the call is the same regardless of where you choose to use it.
+Returning a table from the SSE plugin can be done using the `LOAD ... EXTENSION ...` statement in the Qlik Load script. Read more about the syntax for table load using SSE in the [Qlik Sense help](http://help.qlik.com/en-US/sense/February2018/Subsystems/Hub/Content/Scripting/ScriptRegularStatements/Load.htm) .  
```

Comment:
You can return a table from the SSE plugin  using the `LOAD ... EXTENSION ...` statement in the Qlik Load script.